src/freenet/node/Node.java

```diff
@@ -3166,6 +3172,8 @@ public void start(boolean noSwaps) throws NodeInitException {
 		// Process any data in the extra peer data directory
 		peers.readExtraPeerData();
 
+		((DefaultNodeDiagnostics)nodeDiagnostics).start();
```

Comment:
If you declare the field as `DefaultNodeDiagnostics` you can remove this cast, exactly like you did in `DefaultNodeDiagnostics` with the `DefaultThreadDiagnostics`.